internal/services/grpc/txsvc/v1/tx_query.go

```diff
@@ -28,14 +28,19 @@ func (s *Service) TxQuery(ctx context.Context, req *txpb.TxQueryRequest) (*txpb.
 		return nil, status.Error(codes.Unknown, "failed to query transaction")
 	}
 
-	originalTx := &transactions.Transaction{}
-	if err := originalTx.UnmarshalBinary(cmtResult.Tx); err != nil {
-		logger.Error("failed to deserialize transaction", zap.Error(err))
-		return nil, status.Error(codes.Internal, "failed to deserialize transaction")
+	// cmtResult.Tx can be nil if the transaction is not found
+	var tx *txpb.Transaction
+	var originalTx *transactions.Transaction
```

Comment:
Can omit this decl.  In the conditional block, can declare with `:=` since it's scope is limited.  Only `tx` needed out here.